BUG: groupby by argument raises with mypy when argument is annotated as ``list[str]``

**Describe the bug**

Using ``df.groupby(by=cols)`` where cols is annotated as ``list[str]`` raises a mypy error

```
error: Argument "by" to "groupby" of "DataFrame" has incompatible type "List[str]"; expected "Optional[Union[Tuple[Any, ...], List[Optional[Hashable]], Union[ufunc, Callable[..., Any]], Series[Any], ndarray[Any, Any], Mapping[Optional[Hashable], Any], Index]]"  [arg-type]
``` 

**To Reproduce**
1. Provide a minimal runnable `pandas` example that is not properly checked by the stubs.
2. Indicate which type checker you are using (`mypy` or  `pyright`).
3. Show the error message received from that type checker while checking your example.

```
def _call_groupby(df: pd.DataFrame, cols: list[str]):
    df.groupby(by=cols).count()
```

This also raises when adding:

```
    cols: list[str] = ["col1", "col2"]
    df.groupby(by=cols).sum()
```

to ``test_types_groupby`` in ``test_frame.py``

``list[str]`` should be valid? If passing simply ``["col1", "col2"]`` without an annotation this works

**Please complete the following information:**
 - OS:  ubuntu
 - OS Version 22.04
 - python version 3.10
 - version of type checker mypy 0.971
 - version of installed `pandas-stubs` newest 20220807


**Additional context**
Add any other context about the problem here.

--- a/pandas-stubs/_typing.pyi
+++ b/pandas-stubs/_typing.pyi
@@ -198,7 +198,7 @@ XMLParsers = Literal["lxml", "etree"]
 # Any plain Python or numpy function
 Function = Union[np.ufunc, Callable[..., Any]]
 GroupByObjectNonScalar = Union[
-    tuple, list[Label], Function, Series, np.ndarray, Mapping[Label, Any], Index
+    tuple, list[HashableT], Function, Series, np.ndarray, Mapping[Label, Any], Index
 ]
 GroupByObject = Union[Scalar, GroupByObjectNonScalar]
 

--- a/tests/test_frame.py
+++ b/tests/test_frame.py
@@ -565,6 +565,7 @@ def test_types_groupby() -> None:
     df10: pd.DataFrame = df.groupby("col1").agg(
         new_col=pd.NamedAgg(column="col2", aggfunc="max")
     )
+    # GH#187
     cols: list[str] = ["col1", "col2"]
     check(assert_type(df.groupby(by=cols).sum(), pd.DataFrame), pd.DataFrame)
 

--- a/tests/test_frame.py
+++ b/tests/test_frame.py
@@ -546,13 +546,6 @@ def test_types_groupby() -> None:
     df.groupby(by="col1")
     df.groupby(level="ind")
     df.groupby(by="col1", sort=False, as_index=True)
-
-    cols: list[str] = ["col1", "col2"]
-    df.groupby(by=cols).sum()
-
-    cols_opt: list[str | None] = ["col1", "col2"]
-    df.groupby(by=cols_opt).sum()
-
     df.groupby(by=["col1", "col2"])
 
     df1: pd.DataFrame = df.groupby(by="col1").agg("sum")
@@ -572,6 +565,11 @@ def test_types_groupby() -> None:
     df10: pd.DataFrame = df.groupby("col1").agg(
         new_col=pd.NamedAgg(column="col2", aggfunc="max")
     )
+    cols: list[str] = ["col1", "col2"]
+    df11: pd.DataFrame = df.groupby(by=cols).sum()
+
+    cols_opt: list[str | None] = ["col1", "col2"]
+    df12 : pd.DataFrame = df.groupby(by=cols_opt).sum()
 
 
 # This was added in 1.1.0 https://pandas.pydata.org/docs/whatsnew/v1.1.0.html

--- a/tests/test_frame.py
+++ b/tests/test_frame.py
@@ -546,6 +546,13 @@ def test_types_groupby() -> None:
     df.groupby(by="col1")
     df.groupby(level="ind")
     df.groupby(by="col1", sort=False, as_index=True)
+
+    cols: list[str] = ["col1", "col2"]
+    df.groupby(by=cols).sum()
+
+    cols_opt: list[str | None] = ["col1", "col2"]
+    df.groupby(by=cols_opt).sum()
+
     df.groupby(by=["col1", "col2"])
 
     df1: pd.DataFrame = df.groupby(by="col1").agg("sum")

--- a/tests/test_frame.py
+++ b/tests/test_frame.py
@@ -566,10 +566,10 @@ def test_types_groupby() -> None:
         new_col=pd.NamedAgg(column="col2", aggfunc="max")
     )
     cols: list[str] = ["col1", "col2"]
-    df11: pd.DataFrame = df.groupby(by=cols).sum()
+    check(assert_type(df.groupby(by=cols).sum(), pd.DataFrame), pd.DataFrame)
 
     cols_opt: list[str | None] = ["col1", "col2"]
-    df12: pd.DataFrame = df.groupby(by=cols_opt).sum()
+    check(assert_type(df.groupby(by=cols_opt).sum(), pd.DataFrame), pd.DataFrame)
 
 
 # This was added in 1.1.0 https://pandas.pydata.org/docs/whatsnew/v1.1.0.html

--- a/tests/test_frame.py
+++ b/tests/test_frame.py
@@ -569,7 +569,7 @@ def test_types_groupby() -> None:
     df11: pd.DataFrame = df.groupby(by=cols).sum()
 
     cols_opt: list[str | None] = ["col1", "col2"]
-    df12 : pd.DataFrame = df.groupby(by=cols_opt).sum()
+    df12: pd.DataFrame = df.groupby(by=cols_opt).sum()
 
 
 # This was added in 1.1.0 https://pandas.pydata.org/docs/whatsnew/v1.1.0.html

